cmake/kconfig.cmake

```diff
@@ -104,6 +104,38 @@ if(EXTRA_KCONFIG_OPTIONS)
     )
 endif()
 
+# Set overlay config files list using a list of directories where the
+# overlay files could be found.
+if(DEFINED ENV{OVERLAY_CONFIG_DIRS})
+  set(OVERLAY_CONFIG_DIRS $ENV{OVERLAY_CONFIG_DIRS})
+
+  # We add current directory first so that it is seached first
+  set(overlay_config_dir_list ".:${OVERLAY_CONFIG_DIRS}")
+  string(REPLACE ":" ";" overlay_config_dirs "${overlay_config_dir_list}")
+
+  foreach(f IN ITEMS ${OVERLAY_CONFIG_AS_LIST})
+    if(IS_ABSOLUTE ${f})
+      list(APPEND overlay_config_files ${f})
+    else()
+      set(found 0)
+      foreach(d ${overlay_config_dirs})
+        if(EXISTS ${d}/${f} AND NOT IS_DIRECTORY ${d}/${f})
+          message(STATUS "Overlaying ${d}/${f}")
+          list(APPEND overlay_config_files ${d}/${f})
+          set(found 1)
+          break()
+        endif()
+      endforeach()
+      if(NOT found)
+        # Let the file checker below to report error
```

Comment:
I don't know what this means.

Reply:
What I meant by this comment, is that in about 20-30 lines below in this file, there is a checker that makes sure that the config fragment file exists (search `message(FATAL_ERROR "File not found: ${f}")`). So we do not bail out here if the file is not found as that is checked later in this file anyway.